script/gen/cxx14/all

```diff
@@ -0,0 +1,419 @@
+    filep.write("#include <measurement_kit/ffi.h>\n")
+    filep.write("\n")
+    filep.write("namespace mk {\n")
+    filep.write("namespace cxx14 {\n")
+    filep.write("\n")
+
+def write_ffi_helpers(filep):
+    """ Writes helpers to handle FFI types """
+    filep.write("/// Deletes a mk_task_t pointer\n")
+    filep.write("class TaskDeleter {\n")
+    filep.write("  public:\n")
+    filep.write("    void operator()(mk_task_t *task) noexcept { mk_task_destroy(task); }\n")
+    filep.write("};\n")
+    filep.write("\n")
+    filep.write("/// Unique pointer to mk_task_t\n")
+    filep.write("using TaskUptr = std::unique_ptr<mk_task_t, TaskDeleter>;\n")
+    filep.write("\n")
+    filep.write("/// Deletes a mk_event_t pointer\n")
+    filep.write("class EventDeleter {\n")
+    filep.write("  public:\n")
+    filep.write("    void operator()(mk_event_t *event) noexcept { mk_event_destroy(event); }\n")
+    filep.write("};\n")
+    filep.write("\n")
+    filep.write("/// Unique pointer to mk_event_t\n")
+    filep.write("using EventUptr = std::unique_ptr<mk_event_t, EventDeleter>;\n")
+    filep.write("\n")
+
+def write_events_classes(filep, events):
+    """ Writes event classes on the header file """
+    for key, value in sorted(events.items()):
+        filep.write("/// C++ representation of '%s' event\n" % key)
+        filep.write("class %s {\n" % event_name_to_camel_case(key))
+        filep.write("  public:\n")
+        for name, pattern in value.items():
+            filep.write("    ")
+            if isinstance(pattern, STR):
+                filep.write("std::string %s;\n" % name)
+            elif isinstance(pattern, int):
+                filep.write("int64_t %s = 0;\n" % name)
+            elif isinstance(pattern, float):
+                filep.write("double %s = 0.0;\n" % name)
+            else:
+                raise RuntimeError("unexpected type for: %s" % pattern)
+        filep.write("};\n\n")
+
+def write_events_callbacks(filep, events):
+    """ Writes events callbacks in the header file """
+    for key, value in sorted(events.items()):
+        filep.write("/// Callback taking `const %s &` as unique argument\n" %
+                    event_name_to_camel_case(key))
+        filep.write("using %sCallback = std::function<void(const %s &)>;\n" % (
+            event_name_to_camel_case(key), event_name_to_camel_case(key)))
+        filep.write("\n")
+
+def write_taskinfo_class(filep, settings, sections, events):
+    filep.write("/// Info on a generic task\n")
+    filep.write("class TaskInfo {\n")
+    filep.write("  public:\n")
+    for name in sorted(settings.keys()):
+        value, omit_nl = settings[name], False
+        filep.write("    /// '%s' field of settings\n" % name)
+        if isinstance(value, list):
+            filep.write("    std::vector<std::string> %s;\n" % name)
+            filep.write("\n")
+            filep.write("    /// Allows to append to '%s' using a fluent style\n" % name)
```

Comment:
maybe use "fluent interface" rather than "fluent style" (need to check for the canonical name of the thing)